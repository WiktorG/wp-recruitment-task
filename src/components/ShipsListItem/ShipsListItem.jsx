import React from 'react';
import PropTypes from 'prop-types';
import { v1 as uuid } from 'uuid';

import AddToCartForm from '~/components/AddToCartForm';

import {
    StyledShipsListItem,
    StyledName,
    StyledLabel,
    StyledManufacturers,
    StyledManufacturer,
    StyledPrice,
    StyledFooter,
} from './ShipsListItem.styled';

const Manufacturers = ({ items }) => items.map((item, index) => (
    <React.Fragment
        key={uuid()}
    >
        <StyledManufacturer>
            {item}
        </StyledManufacturer>
        {(index + 1 !== items.length) ? ' and ' : ''}
    </React.Fragment>
));

const ShipsListItem = ({
    name,
    manufacturers,
    costInCredits,
}) => {
    const avaliable = costInCredits !== null;

    return (
        <StyledShipsListItem
            data-testid="ListItem"
        >
            <StyledName>
                {name}
            </StyledName>
            <StyledManufacturers>
                <StyledLabel>
                    by
                    {' '}
                </StyledLabel>
                <Manufacturers items={manufacturers} />
            </StyledManufacturers>
            <StyledFooter>
                {avaliable ? (
                    <>
                        <StyledPrice>
                            <StyledLabel>
                                Price:
                            </StyledLabel>
                            {costInCredits}
                        </StyledPrice>
                        <AddToCartForm />
                    </>
                ) : (
                    <StyledPrice>
                        Currently unavaliable
                    </StyledPrice>
                )}
            </StyledFooter>
        </StyledShipsListItem>
    );
};

ShipsListItem.propTypes = {
    name: PropTypes.string.isRequired,
    manufacturers: PropTypes.arrayOf(PropTypes.string).isRequired,
    costInCredits: PropTypes.number,
};

ShipsListItem.defaultProps = {
    costInCredits: null,
};

export default ShipsListItem;
