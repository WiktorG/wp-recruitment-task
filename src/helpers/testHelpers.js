import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '~/redux/store';


export const renderWithReduxProvider = (component) => render(
    <Provider store={store}>
        {component}
    </Provider>,
);
