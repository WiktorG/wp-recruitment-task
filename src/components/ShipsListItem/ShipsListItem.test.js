import React from 'react';
import { screen } from '@testing-library/react';
import { renderWithReduxProvider } from '~/helpers/testHelpers';

import ShipsListItem from './ShipsListItem';

test('component should render', async () => {
    renderWithReduxProvider(
        <ShipsListItem
            name="SpaceShip"
            manufacturers={['Manufacturer1', 'Manufacturer2']}
            costInCredits={100}
        />,
    );
    const skeletonNode = screen.getByTestId('ListItem');
    expect(skeletonNode).toMatchSnapshot();
});
