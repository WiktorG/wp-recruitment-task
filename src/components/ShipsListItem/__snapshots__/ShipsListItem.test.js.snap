// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component should render 1`] = `
<li
  class="sc-AxgMl kKHxzH"
  data-testid="ListItem"
>
  <span
    class="sc-AxheI bvrtTr"
  >
    SpaceShip
  </span>
  <div
    class="sc-AxmLO hccitz"
  >
    <span
      class="sc-Axmtr jkUeyX"
    >
      by
       
    </span>
    <span
      class="sc-fzozJi eCoiWl"
    >
      Manufacturer1
    </span>
     and 
    <span
      class="sc-fzozJi eCoiWl"
    >
      Manufacturer2
    </span>
    
  </div>
  <div
    class="sc-fzpans eWWpQg"
  >
    <span
      class="sc-fzoLsD fNmcxP"
    >
      <span
        class="sc-Axmtr jkUeyX"
      >
        Price:
      </span>
      100
    </span>
    <form
      class="sc-AxjAm sHBsz"
      data-testid="Form"
    >
      <div
        class="sc-AxirZ jlZVKK"
      >
        <button
          class="sc-AxiKw dbsbqc"
        >
          -
        </button>
        <input
          class="sc-AxhCb hqwelk"
          data-testid="Input"
          type="number"
          value="1"
        />
        <button
          class="sc-AxiKw dbsbqc"
        >
          +
        </button>
      </div>
      <button
        class="sc-AxhUy kaApbH"
        data-testid="Button"
        type="submit"
      >
        Add to cart
      </button>
    </form>
  </div>
</li>
`;
